optimum/onnxruntime/quantization.py

```diff
@@ -0,0 +1,262 @@
+                exists.
+            revision(:obj:`str`, `optional`):
+                The specific version to use. It can be a branch name, a tag name, or a commit id, since we use a
+                git-based system for storing models and other artifacts on huggingface.co, so ``revision`` can be any
+                identifier allowed by git.
+        """
+        config_kwargs_default = [
+            ("cache_dir", None),
+            ("force_download", False),
+            ("resume_download", False),
+            ("revision", None),
+        ]
+        config_kwargs = {name: kwargs.get(name, default_value) for (name, default_value) in config_kwargs_default}
+        self.cache_dir = config_kwargs.get("cache_dir")
+        self.model_name_or_path = model_name_or_path
+        self.output_dir = output_dir if isinstance(output_dir, Path) else Path(output_dir)
+        self.model_path = self.output_dir.joinpath("model.onnx")
+        self.quant_model_path = generate_identified_filename(self.model_path, "-quantized")
+        if not isinstance(ort_config, ORTConfig):
+            ort_config = ORTConfig.from_pretrained(ort_config, **config_kwargs)
+        self.ort_config = ort_config
+        self.quantization_approach = ort_config.quantization_approach
+        self.activation_type = Q_TYPE.get(ort_config.activation_type, QuantType.QUInt8)
+        self.weight_type = Q_TYPE.get(ort_config.weight_type, QuantType.QUInt8)
+        self.quant_format = Q_FORMAT.get(ort_config.quant_format, QuantFormat.QOperator)
+        self.calibrate_method = CALIB_METHOD.get(ort_config.calibration_method, CalibrationMethod.MinMax)
+        self.calib_dataset = calib_dataset
+        self.dataset_name = dataset_name
+        self.dataset_config_name = dataset_config_name
+        self.data_files = data_files
+        self.preprocess_function = preprocess_function
+        self.onnx_config = None
+        self.feature = feature
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_name_or_path)
+        self.model = FeaturesManager.get_model_from_feature(self.feature, self.model_name_or_path)
+
+    def export(self) -> None:
+        """
+        Load and export a model to an ONNX Intermediate Representation (IR).
+        """
+        model_type, model_onnx_config = FeaturesManager.check_supported_model_or_raise(
+            self.model, feature=self.feature
+        )
+        self.onnx_config = model_onnx_config(self.model.config)
+        opset = self.onnx_config.default_onnx_opset if self.ort_config.opset is None else self.ort_config.opset
+        _ = export(self.tokenizer, self.model, self.onnx_config, opset, self.model_path)
+
+    def fit(self) -> None:
+        """
+        Load and export a model to an ONNX Intermediate Representation (IR) after applying the specified quantization
+        approach.
+        """
+        self.export()
+        if self.quantization_approach == ORTQuantizationMode.DYNAMIC.value:
```

Comment:
I would suggest you rely on enum comparison rather than string comparison (`.value`).

You can do: 

```python
self.quantization_approach = ORTQuantizationMode(quantization_approach)
...
if self.quantization_approach == ORTQuantizationMode.DYNAMIC
```